server/src/main/java/io/crate/metadata/GeoReference.java

```diff
@@ -86,7 +81,12 @@ public GeoReference(ReferenceIdent ident,
             isDropped,
             defaultExpression
         );
-        this.geoTree = Objects.requireNonNullElse(geoTree, DEFAULT_TREE);
+        this.geoTree = Objects.requireNonNullElse(geoTree, GEOHASH_PREFIX_REE);
+        if (BKD_TREE.equals(this.geoTree) && (precision != null || treeLevels != null || distanceErrorPct != null)) {
+            throw new IllegalArgumentException(
+                "The parameters precision, tree_levels, and distance_error_pct are not applicable to BKD tree indexes."
+            );
+        }
```

Comment:
I suggest to move this check to analyzer (we create `GeoReference` in `TableElementsAnalyzer.RefBuilder.build`). 

Then we can replace `DDLIntegrationTest.testSettingsNotApplicableToBkdTreeIndexes` by more lightweight tests cases in `AlterTableAddColumnAnalyzerTest` and `CreateAlterTableStatementAnalyzerTest`

Reply:
yep, let's keep the check in ctor but rewrite as  2 unit test - create table and add column.

Reply:
I replaced `DDLIntegrationTest.testSettingsNotApplicableToBkdTreeIndexes` with `GeoReferenceTest.testSettingsNotApplicableToBkdTreeIndexes`.